tests/Unit/ProtoTests/Kms/out/samples/V1/KeyManagementServiceClient/import_crypto_key_version.php

```diff
@@ -0,0 +1,82 @@
+use Google\Cloud\Kms\V1\KeyManagementServiceClient;
+
+/**
+ * Import wrapped key material into a
+ * [CryptoKeyVersion][google.cloud.kms.v1.CryptoKeyVersion].
+ *
+ * All requests must specify a [CryptoKey][google.cloud.kms.v1.CryptoKey]. If
+ * a [CryptoKeyVersion][google.cloud.kms.v1.CryptoKeyVersion] is additionally
+ * specified in the request, key material will be reimported into that
+ * version. Otherwise, a new version will be created, and will be assigned the
+ * next sequential id within the [CryptoKey][google.cloud.kms.v1.CryptoKey].
+ *
+ * @param string                    $formattedParent Required. The [name][google.cloud.kms.v1.CryptoKey.name] of the
+ *                                                   [CryptoKey][google.cloud.kms.v1.CryptoKey] to be imported into.
+ *
+ *                                                   The create permission is only required on this key when creating a new
+ *                                                   [CryptoKeyVersion][google.cloud.kms.v1.CryptoKeyVersion].
+ * @param CryptoKeyVersionAlgorithm $algorithm       Required. The
+ *                                                   [algorithm][google.cloud.kms.v1.CryptoKeyVersion.CryptoKeyVersionAlgorithm]
+ *                                                   of the key being imported. This does not need to match the
+ *                                                   [version_template][google.cloud.kms.v1.CryptoKey.version_template] of the
+ *                                                   [CryptoKey][google.cloud.kms.v1.CryptoKey] this version imports into.
+ * @param string                    $importJob       Required. The [name][google.cloud.kms.v1.ImportJob.name] of the
+ *                                                   [ImportJob][google.cloud.kms.v1.ImportJob] that was used to wrap this key
+ *                                                   material.
+ */
+function import_crypto_key_version_sample(string $formattedParent, CryptoKeyVersionAlgorithm $algorithm, string $importJob)
+{
+    $keyManagementServiceClient = new KeyManagementServiceClient();
+    
+    try {
+        /** @var CryptoKeyVersion $response */
+        $response = $keyManagementServiceClient->importCryptoKeyVersion($formattedParent, $algorithm, $importJob);
+        printf('Response data: %s' . PHP_EOL, $response->serializeToJsonString());
+    } catch (ApiException $ex) {
+        printf('Call failed with message: %s' . PHP_EOL, $ex->getMessage());
+    }
+}
+
+/**
+ * Helper to execute the sample.
+ *
+ * TODO(developer): Replace sample parameters before running the code.
+ */
+function callSample()
+{
+    $formattedParent = KeyManagementServiceClient::cryptoKeyName('[PROJECT]', '[LOCATION]', '[KEY_RING]', '[CRYPTO_KEY]');
+    $algorithm = CryptoKeyVersionAlgorithm::CRYPTO_KEY_VERSION_ALGORITHM_UNSPECIFIED;
+    $importJob = 'import_job';
+    
+    import_crypto_key_version_sample($formattedParent, $algorithm, $importJob);
```

Comment:
This is an interesting sample. I think in cases like this, I'd prefer to move this into the sample itself, and make the above place holders the sample params.
```suggestion
    $project = 'project';
    $location = 'location';
    $keyRing = 'key_ring';
    $cryptoKey = 'crypto_key';
    $algorithm = CryptoKeyVersionAlgorithm::CRYPTO_KEY_VERSION_ALGORITHM_UNSPECIFIED;
    $importJob = 'import_job';
    
    import_crypto_key_version_sample($project, $location, $keyRing, $cryptoKey, $algorithm, $importJob);
```

Reply:
I'm glad you thought about this too - at first I tried this out, but there were a couple challenges I ran into that made me revert back to this existing approach.

- In the docblock for the main sample, project/location/keyRing etc. wouldn't have any descriptions we could pull in, whereas if we keep the formatted string as the parameter it allows us to leverage the description from the protos: https://github.com/googleapis/gapic-generator-php/blob/e68b44cea965f678731f7f57e2044acfc01ab25b/tests/Unit/ProtoTests/Kms/service.proto#L742-L746
- Some RPCs (at least, outlined in the library service) have multiple formatting methods, which can lead to a really large method signature if we split out each individual component of the resource. It also means we have to take some liberty to programmatically determine what the variable name is and avoid any collisions. I found this was leading to really long, and not entirely helpful variable names - for example: https://github.com/googleapis/gapic-generator-php/blob/e68b44cea965f678731f7f57e2044acfc01ab25b/tests/Unit/ProtoTests/BasicDiregapic/library_rest.proto#L573-L582

Reply:
We addressed this in the meeting and decided the approach here is preferred, as long as we add a `@see` tag with the resource name helper to the sample PHPDoc.